src/components/ha-copy-textfield.ts

```diff
@@ -0,0 +1,132 @@
+import { customElement, property, state } from "lit/decorators";
+import {
+  type PropertyValues,
+  css,
+  type CSSResultGroup,
+  html,
+  LitElement,
+  nothing,
+} from "lit";
+import { mdiContentCopy, mdiEye, mdiEyeOff } from "@mdi/js";
+import { haStyle } from "../resources/styles";
+
+import "./ha-button";
+import "./ha-icon-button";
+import "./ha-svg-icon";
+import "./ha-textfield";
+import type { HomeAssistant } from "../types";
+import { copyToClipboard } from "../common/util/copy-clipboard";
+import { showToast } from "../util/toast";
+import type { HaTextField } from "./ha-textfield";
+
+@customElement("ha-copy-textfield")
+export class HaCopyTextfield extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: "value" }) public value!: string;
+
+  @property({ attribute: "masked_value" }) public maskedValue?: string;
+
+  @property({ attribute: "label" }) public label?: string;
+
+  @state() private _showMasked = false;
+
+  protected firstUpdated(_changedProperties: PropertyValues) {
+    super.firstUpdated(_changedProperties);
+
+    if (this.maskedValue) {
+      this._showMasked = true;
+    }
+  }
```

Comment:
This runs after the 1. render. I think it makes more sense to use `willUpdate`  and `if (!this.hasUpdated)`

Reply:
I can actually omit this completely after a second look, I can set `_showMasked` to `true` by default.

I later check if `maskedValue` is set